src/server.c

```diff
@@ -5318,48 +5319,44 @@ int main(int argc, char **argv) {
                 exit(1);
             }
         }
-
-        /* First argument is the config file name? */
-        if (argv[j][0] != '-' || argv[j][1] != '-') {
-            configfile = argv[j];
-            server.configfile = getAbsolutePath(configfile);
-            /* Replace the config file in server.exec_argv with
-             * its absolute path. */
-            zfree(server.exec_argv[j]);
-            server.exec_argv[j] = zstrdup(server.configfile);
-            j++;
+        /* Parse command line options
+         * Precedence wise, File, stdin, explicit options -- last config is the one that matters.
+         *
+         * First argument is the config file name? */
+        if (argv[1][0] != '-') {
+            /* Replace the config file in server.exec_argv with its absolute path. */
+            server.configfile = getAbsolutePath(argv[1]);
+            zfree(server.exec_argv[1]);
+            server.exec_argv[1] = zstrdup(server.configfile);
+            j = 2; // Skip this arg when parsing options
+        }
+        /* Last argument - Should we read config from stdin? */
+        if (argv[argc-1][0] == '-' && argv[argc-1][1] == '\0') {
+            config_from_stdin = 1;
         }
-
         /* All the other options are parsed and conceptually appended to the
          * configuration file. For instance --port 6380 will generate the
-         * string "port 6380\n" to be parsed after the actual file name
-         * is parsed, if any. */
-        while(j != argc) {
+         * string "port 6380\n" to be parsed after the actual config file
+         * and stdin input are parsed (if they exist). */
+        while(j < (argc - (config_from_stdin ? 1 : 0))) {
             if (argv[j][0] == '-' && argv[j][1] == '-') {
                 /* Option name */
-                if (!strcmp(argv[j], "--check-rdb")) {
-                    /* Argument has no options, need to skip for parsing. */
-                    j++;
-                    continue;
```

Comment:
i couldn't find when / what this was doing. can you give me a pointer?

Reply:
Sure -- see commit 145473acc5798a499a0f37e42df48a014a3955c1
and commit 7d1e15808422a1233280440566a5c684de7aa92f (which removed support)